Modules/_decimal/_decimal.c

```diff
@@ -3241,11 +3243,24 @@ dec_format(PyObject *dec, PyObject *args)
             /* NUL fill character: must be replaced with a valid UTF-8 char
                before calling mpd_parse_fmt_str(). */
             replace_fillchar = 1;
-            fmt = dec_strdup(fmt, size);
-            if (fmt == NULL) {
+            fmt = fmt_copy = dec_strdup(fmt, size);
+            if (fmt_copy == NULL) {
                 return NULL;
             }
-            fmt[0] = '_';
+            fmt_copy[0] = '_';
+        }
+        char *z_start = strchr(fmt, 'z');
+        if (z_start != NULL) {
+            no_neg_0 = 1;
+            size_t z_index = z_start - fmt;
+            if (fmt_copy == NULL) {
+                fmt = fmt_copy = dec_strdup(fmt, size);
+                if (fmt_copy == NULL) {
+                    return NULL;
+                }
+            }
+            memmove(fmt_copy + z_index, fmt_copy + z_index + 1, size - z_index);
+            size -= 1;
```

Comment:
if PR #29438 lands, then detection of the "z" option can be incorporated into our forked `mpd_parse_fmt_str_ex()`, rather than this preprocessing of the format string

Reply:
I think it's the case that one way or the other, the format spec is going to need to be completely parsed to figure out if a given 'z' is used for no_neg_0.